File: InitialTrainingLibrary/InitialTrainingLibrary/mg/Chess/Figure.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using InitialTrainingLibrary.Interfaces.chess;

namespace InitialTrainingLibrary.mg.Chess
{
    abstract class Figure :IFigure
    {
        protected FigureKind figureKind;
        protected Coordinates coordinates;
        protected bool isWhite;

        public bool IsFigureWhite()
        {
            return isWhite;
        }

        public ICoordinates GetCoordinates()
        {
            return coordinates;
        }

        public FigureKind GetFigureKind()
        {
            return figureKind;
        }

        public bool Move(ICoordinates newCoordinates)
        {
            throw new NotImplementedException();
        }
    }
}

File: Eagles/TrainingSix/Chess/Figure.php
<?php


	abstract class Figure implements IFIgure
	{
		protected $shape;
		
		protected function __construct($shape)
		{
			$this->shape = $shape;
		}
		
		abstract function Move($coordinates);
		
		abstract function GetColour();
		
		function GetShape()
		{
			return $this->shape;
		}
	}